spec/models/lead_provider_delivery_partnership_spec.rb

```diff
@@ -51,9 +51,21 @@
   end
 
   describe "declarative touch" do
-    let(:instance) { FactoryBot.create(:lead_provider_delivery_partnership, delivery_partner: target) }
-    let(:target) { FactoryBot.create(:delivery_partner) }
+    let(:instance) { FactoryBot.create(:lead_provider_delivery_partnership, delivery_partner:) }
+    let(:delivery_partner) { FactoryBot.create(:delivery_partner) }
+
+    context "target delivery_partner" do
+      let(:target) { delivery_partner }
+
+      it_behaves_like "a declarative touch model", on_event: %i[create destroy], timestamp_attribute: :api_updated_at
+    end
+
+    context "target school_partnerships" do
+      let!(:school_partnership) { FactoryBot.create(:school_partnership, lead_provider_delivery_partnership: instance) }
+      let(:target) { school_partnership }
+      let(:new_value) { FactoryBot.create(:delivery_partner).id }
```

Comment:
Are we able to keep this explicit as with how we do it in schools? I'm just thinking if we added any other attribute to the touch model it'll not work as expected and might be confusing? I think this is a bit clearer (but maybe you can't pass an `id` into a factory? never tried...):

```
def will_change_attribute(attribute_to_change:, new_value:)
  FactoryBot.create(:delivery_partner, id: new_value) if attribute_to_change == :delivery_partner_id
end
```

Reply:
It would be better to allow us to explicitly create the new value on the spec as needed, instead of always randomly generating. Added this:

```ruby
      def generate_new_value(attribute_to_change:)
        case attribute_to_change
        when :lead_provider_delivery_partnership_id
          FactoryBot.create(:lead_provider_delivery_partnership).id
        end
      end
```
Should maintain the existing behaviour.